### ISSUE TITLE
Git hooks for contributors

### ISSUE THREAD

**Issue by Mandera** [ManderaGeneral/generalpackager#80]
Since the `.git` folder isn't versioned for security reasons to execute arbitrary code we have an issue; `pre-commit` and `pre-push` hooks won't be there when forked or cloned.

## Options
- Rely on `generate.py` to create all hooks for us - already implemented
  - Might get tricky to merge with additional checks - The pre-commit package has this [nice config file](https://github.com/deepset-ai/haystack/blob/main/.pre-commit-config.yaml). Not sure how that works under the hood, might also be easy 
- Rely on pre-commit package and remove [git hook definitions](https://github.com/ManderaGeneral/generalpackager/blob/master/generalpackager/api/shared/files/definitions/pre_push_hook.py) from Packager
  - Implement the `pre-commit install` command somewhere, inside `generate_localfiles` or make a new method probably coupled to https://github.com/ManderaGeneral/generalpackager/blob/bd361ec6f4b4d3fad71e2ddf6b66be47666e3916/generalpackager/packager_files.py#L7
  - Could add the `generate_localfiles` step in `.pre-commit-config.yaml` as a pre-commit repo perhaps?

**Comment by Mandera** [comment]
Might not be such a good idea to have generate.py to setup the dev environment, because it would require generalpackager to be installed.
Although I guess generalpackager could be put in every repos [dev] dependencies. Having circular dependency through the optional [dev] might be fine. These dev deps should be ignored by the mermaid graph too

`generate.py` is just something I made because I got tired of generating the workflow by hand, then it turned out to be pretty useful to keep git hooks in sync as well. Also nice to make sure the readme has been generated properly

It's basically what happens in the sync stage of the workflow as well. Running it once will enable the pre-push hook to run it automatically, but only important (non-aesthetic) files

To summarize, `generate_localfiles` is used in a few places:
- Manually by `generate.py`
- In the pre push hook to halt if any non-aesthetic file was changed
- In the CI unittest step (Might not be needed)
- In the CI sync stage

The [linting](https://github.com/ManderaGeneral/generalimport/issues/23) should probably be included in all of these places
